### DISCUSSION TITLE
when loading a npm module dynamically the app crashes.

### DISCUSSION THREAD

**Discussion by handrei** [vercel/next.js#15738] [Help] [answered]
const { ReactTinyLink } = dynamic(() => import('react-tiny-link'), {
  ssr: false,
});

am I not allowed to destruct like this?

the error I get is `Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.`

**Comment by jamesmosier** [comment, marked as answer]
You could try importing it like this:

```js
const ReactTinyLink = dynamic(() =>
  import('react-tiny-link').then((mod) => mod.ReactTinyLink),
  {ssr: false}
)
```

[next/dynamic named export docs](https://nextjs.org/docs/advanced-features/dynamic-import#with-named-exports)

**Reply by handrei** [reply]
thank you, this is working, but not properly. now i have another issue:

`Unhandled Runtime Error
TypeError: Failed to construct 'URL': Invalid URL`

**Reply by handrei** [reply]
i don't know if it's from module, or from my code :(

**Reply by handrei** [reply]
its from react-tiny-link module. thanks for the help 👍

**Reply by ManalLiaquat** [reply]
Thanks for the help @jamesmosier 😃
